pkg/traceutil/trace.go

```diff
@@ -0,0 +1,60 @@
+package traceutil
+
+import (
+	"bytes"
+	"fmt"
+	"time"
+
+	"github.com/coreos/pkg/capnslog"
+	"go.uber.org/zap"
+)
+
+var (
+	plog = capnslog.NewPackageLogger("go.etcd.io/etcd", "trace")
+)
+
+type Trace struct {
+	operation string
+	startTime time.Time
+	steps     []step
+}
+
+type step struct {
+	time time.Time
+	msg  string
+}
+
+func New(op string) *Trace {
+	return &Trace{operation: op, startTime: time.Now()}
+}
+
+func (t *Trace) Step(msg string) {
+	t.steps = append(t.steps, step{time: time.Now(), msg: msg})
+}
+
+// Log dumps all steps in the Trace
+func (t *Trace) Log(lg *zap.Logger) {
+
+	var buf bytes.Buffer
+
+	buf.WriteString(fmt.Sprintf("The tracing of %v request:\n", t.operation))
```

Comment:
Let's tune the trace log output to be compact and machine parse-able. E.g. keep the number of lines of output to a reasonable minimum and minimize filler text. E.g.:

```
Trace "Obfuscating Quigley Matrix" (duration: <trace-time>, start: <start-time>)
Step "Reticulating Splines" (duration: <step-time>)
...
```